app/interfaces/api/logger.rb

```diff
@@ -35,13 +42,21 @@ def response_body
       return if @app_response.blank?
 
       JSON.parse(@app_response[2].first)
-    rescue JSON::ParserError
-      Rails.logger.error "JSON::Parser error parsing: \n#{@app_response[2].first}"
-    end
-
-    def log_api(api_type, data)
-      api_request = { log_type: api_type, timestamp: Time.zone.now }.merge!(data)
-      Rails.logger.info api_request.to_json
+    rescue JSON::ParserError => e
+      log_api(:error,
+              'response-body',
+              "Error parsing API response body: \n#{@app_response[2].first}",
+              e)
+    end
+
+    def log_api(level, type, data, error = nil)
+      LogStuff.send(
```

Comment:
Is it better to pass a hash of arguments to make it more readable and remove any reliance on the order. Or perhaps that method is not going to be reused outside this context anyway and the params are very clear - so maybe not necessary. It was just a thought.

`(level: level, type: type, data: data, error = nil)`

Reply:
I think that might be a bit overkill here? Due to how Grape Middleware works, this is unlikely to be exported to any other areas of the application. It's just an abstraction used in this specific file.

Reply:
That makes sense @AlexRyanShep. This is mostly me just trying to be thorough in a code review and making sure I understand it fully.